test/bugs/gh-1116.js

```diff
@@ -0,0 +1,128 @@
+// https://github.com/swagger-api/swagger-js/issues/1116
+import { buildRequest } from '../../src/execute';
+
+describe('test accept header for oas 3', () => {
+  const spec = {
+    openapi: '3.0.0',
+    servers: [
+      {
+        url: 'https://test.com/v1',
+      },
+    ],
+    paths: {
+      '/getMultiple': {
+        get: {
+          operationId: 'getMultiple',
+          responses: {
+            200: {
+              content: {
+                'application/xml': {
+                  schema: {
+                    type: 'array',
+                    items: 'string',
+                  },
+                },
+                'application/json': {
+                  schema: {
+                    type: 'array',
+                    items: 'string',
+                  },
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  };
+
+  test('oas 3: should generate accept header according to defined responses', () => {
+    const req = buildRequest({
+      spec,
+      operationId: 'getMultiple',
+    });
+
+    expect(req).toEqual({
+      url: 'https://test.com/v1/getMultiple',
+      method: 'GET',
+      credentials: 'same-origin',
+      headers: {
+        accept: 'application/xml, application/json',
+      },
+    });
+  });
+
+  test('oas 3: should allow to override accept header according to defined responses', () => {
+    const req = buildRequest({
+      spec,
+      operationId: 'getMultiple',
+      responseContentType: 'application/xml',
+    });
+
+    expect(req).toEqual({
+      url: 'https://test.com/v1/getMultiple',
+      method: 'GET',
+      credentials: 'same-origin',
+      headers: {
+        accept: 'application/xml',
+      },
+    });
+  });
+});
+
+describe('test accept header for swagger 2', () => {
```

Comment:
Can you please incorporate these tests into https://github.com/swagger-api/swagger-js/blob/master/test/swagger2/execute/build-request.js?

Reply:
Can we please eliminate this new file and incorporate these tests into  https://github.com/swagger-api/swagger-js/blob/master/test/swagger2/execute/build-request.js (as suggested in previous comment) ? I'd like to avoid practice of creating any new tests under `/test/bugs` as it creates fragmentation within the tests. I'd like to keep all the related tests together.